lib/version.js

```diff
@@ -88,8 +89,82 @@ function setCommand(version, pkg) {
 function set(v, project) {
   v = semver(v.toString());
   project.version(v);
+  setBlipIfOpted(project);
+}
+
+function setBlipIfOpted(project) {
+  var opted = false;
+  var blipScript = false;
+  var scriptPhases = {
+    'preinstall': project.scripts('preinstall'),
+    'install': project.scripts('install'),
+    'postinstall': project.scripts('postinstall')
+  };
+  var availPhases = [];
+  //check if the package opted for blip
+  hasOpted(project, scriptPhases, availPhases, opted, blipScript, function(hasOpted, hasBlipScript) {
+    opted = hasOpted;
+    blipScript = hasBlipScript;
+  });
+
+  if (opted && !blipScript) { //when package opted for Blip but doesn't have blip in scripts
+    if (availPhases.length) {
+      project.scripts(availPhases[0], 'node scripts/sl-blip.js'); //use first available script phase
+    } else { //if scripts exist, append sl-blip to existing postinstall script
+      var value = scriptPhases['postinstall'].toString() + ' && node scripts/sl-blip.js';
+      project.scripts('postinstall', value);
+    }
+    updateBlipScript(project);
+  }
+  if (opted && blipScript) { //when package opted for blip & have blip in scripts, update sl-blip.js file
+    updateBlipScript(project);
+  }
+}
+
+function hasOpted(project, scriptPhases, availPhases, opted, blipScript, cb) {
+  //check if optionalDependencies has sl-blip, if yes, delete sl-blip
   if (project.optionalDep('sl-blip')) {
-    var blipUrl = 'http://blip.strongloop.com/' + project.nameVer();
-    project.optionalDep('sl-blip', blipUrl);
+    project.optionalDep('sl-blip', '');
+    opted = true;
+  }
+  //check if scripts has sl-blip setup
+  for (var phase in scriptPhases) {
+    if (scriptPhases.hasOwnProperty(phase) && !scriptPhases[phase]) {
+      availPhases.push(phase);
+    } else {
+      hasBlipScript(scriptPhases[phase]) ? opted = blipScript = true : opted;
+    }
+  }
+  cb(opted, blipScript);
+}
+
+function hasBlipScript(phaseVal) {
+  var reg = /sl-blip/i;
+  if (reg.test(phaseVal.toString())) {
+    return true;
+  }
+  return false;
+}
+
+function updateBlipScript(project) {
+  var SCRIPTS = path.resolve(project.rootPath, 'scripts'); //path to scripts dir of project
+  var SL_BLIP = path.resolve(SCRIPTS, 'sl-blip.js'); //path to sl-blip.js of project
+  var blipjs = fs.readFileSync(path.resolve(__dirname, 'sl-blip.js'), 'utf-8'); //latest sl-blip.js from strong-tools/lib/
+  
+  //if scripts dir doesn't exist, create one
+  try {
+    fs.mkdirSync(SCRIPTS);
+  } catch (e) {
+    if (e.code === 'EEXIST') {
```

Comment:
A conditional where the action only happens in the else condition is an anti-pattern. It would be better written as:
```js
if (e.code !== 'EEXIST') {
  console.error('Failed to create directory: ', e);
}
```